users/rupa/wrappers.h

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2020 rupa <rupa@lrrr.us> @rupa
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+#include "rupa.h"
+
+#if (!defined(LAYOUT) && defined(KEYMAP))
+#    define LAYOUT KEYMAP
+#endif
+
+#define LOWER   MO(_LOWER)
+#define RAISE   MO(_RAISE)
+#define OS_RGUI OSM(MOD_RGUI)
+#define OS_RALT OSM(MOD_RALT)
+#define OS_RCTL OSM(MOD_RCTL)
+#define OS_RSFT OSM(MOD_RSFT)
+
+#define G_LWR   LT(_LOWER, KC_G)
+
+#if defined(UNICODEMAP_ENABLE)
+#    define CSHAPES XP(CCIR,CKEY)
+#    define CUIDADO XP(CUI,HAS)
+#    define NOPENAH XP(NOPE,STOP)
+#    define MUSIC_A XP(M4,M8)
+#    define MUSIC_B XP(M8B,M16)
+#    define SMILE   XP(SMB,SMW)
+#    define STARS   XP(STB,STW)
+#    define YEPYEP  XP(CHEK,HUN)
+#endif
+
+/*
+Since our quirky block definitions are basically a list of comma separated
+arguments, we need a wrapper in order for these definitions to be
+expanded before being used as arguments to the LAYOUT_xxx macro.
+*/
+
+// clang-format off
+#define KEYMAP_wrapper(...)                 LAYOUT(__VA_ARGS__)
+#define LAYOUT_wrapper(...)                 LAYOUT(__VA_ARGS__)
+#define LAYOUT_65_wrapper(...)              LAYOUT_65(__VA_ARGS__)
```

Comment:
Any reason to use layout wrappers, rather than the `65_ansi` community layout?

Reply:
not 100% sure, I'm following @drashna's pattern without understanding much about other options. stealing from the greats as is were :)

Im not using any of the ones above (even in not-yet-submitted keyboards) - my understanding is that these are required per above comment, and we're defining ones we'd expect to reuse across several keyboards, as opposed to defining it in multiple keymap.c files. @drashna's wrappers have similar defines, but for e.g. his moonlander keymap, he just defines the layout wrappers in the keyboard.c file, since it uses its own LAYOUT_moonlander